### PR TITLE
feat: Implements queuing full voice channels

### PR DESCRIPTION
**Pull request by hbina**
(Garbage) implementation of this feature.
Still trying to figure out how to do this sanely

See #7

Signed-off-by: Hanif Bin Ariffin <hanif.ariffin.4326@gmail.com>

### CODE REVIEW

File: src/managers/roommanagerimpl.ts [hunk 1]
```diff
@@ -81,4 +86,31 @@ export class RoomManagerImpl implements RoomManager {
       });
     }
   }
+
+  public addUserToQueue(user: Discord.User, channel: VoiceChannel) {
+    this.userQueue.push({ user, channel });
```
Comment (cglotr) [inline src/managers/roommanagerimpl.ts]: This will be potentially contain duplicates?

File: src/types/channelqueue.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import Discord from 'discord.js';
+import { VoiceChannel } from './voicechannel';
+
+export type ChannelQueue = {
+  user: Discord.User;
```
Comment (cglotr) [inline src/types/channelqueue.ts]: I think better to only store the user id & fetch it on demand when we want to.

File: src/main.ts [hunk 3]
```diff
@@ -340,6 +366,7 @@ client.on('voiceStateUpdate', (oldState, newState) => {
         newState.channel.members.size,
       );
     }
+    roomManager.workOnQueues();
```
Comment (cglotr) [inline src/main.ts]: I think we should move this to a cron process.

File: src/managers/roommanagerimpl.ts [hunk 4]
```diff
@@ -81,4 +86,31 @@ export class RoomManagerImpl implements RoomManager {
...
+  }
+
+  public workOnQueues() {
+    this.loggerService.info(`queues:${JSON.stringify(this.userQueue)}`);
+    const nextQueue: ChannelQueue[] = [];
```
Comment (cglotr) [inline src/managers/roommanagerimpl.ts]: Should we have `n` number of queues, one for each tracked room?

**Review by cglotr — CHANGES_REQUESTED** [review]

**Comment by hbina** [comment]
@cglotr applied the requested changes. The only thing left is prolly checking edge cases and adding unit tests.

File: src/managers/queuemanagerimpl.test.ts [hunk 5]
```diff
@@ -0,0 +1,103 @@
+import { DiscordGuild } from '../discord/discordguild';
```
Comment (hbina) [inline src/managers/queuemanagerimpl.test.ts]: This is the _wrong_ test.

**Review by cglotr — APPROVED** [review]
